plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -779,28 +764,31 @@ private void assertFileSystemAccesses(Session session, @Language("SQL") String q
     {
         assertUpdate("CALL system.flush_metadata_cache()");
 
-        trackingFileSystemFactory.reset();
         getDistributedQueryRunner().executeWithPlan(session, query);
-        assertMultisetsEqual(getOperations(), expectedAccesses);
+        List<SpanData> spanData = getDistributedQueryRunner().getSpans();
+        assertMultisetsEqual(getOperations(spanData), expectedAccesses);
     }
 
-    private Multiset<FileOperation> getOperations()
+    private Multiset<FileOperation> getOperations(List<SpanData> spanData)
     {
-        return trackingFileSystemFactory.getOperationCounts()
-                .entrySet().stream()
-                .filter(entry -> {
-                    String path = entry.getKey().location().path();
+        return spanData.stream()
+                .filter(TestDeltaLakeFileOperations::isFileSystemSpan)
+                .filter(span -> {
+                    String path = requireNonNull(span.getAttributes().get(FILE_LOCATION), "FILE_LOCATION attribute is empty");
                     return !path.endsWith(".trinoSchema") && !path.contains(".trinoPermissions");
                 })
-                .flatMap(entry -> nCopies(entry.getValue(), FileOperation.create(
-                        entry.getKey().location().path(),
-                        entry.getKey().operationType())).stream())
+                .map(span -> FileOperation.create(requireNonNull(span.getAttributes().get(FILE_LOCATION), "FILE_LOCATION attribute is empty"), span.getName()))
```

Comment:
This null check is already done above. Remove it here, and maybe move it into a `peek()` call above. Or add a utility method to extract it.